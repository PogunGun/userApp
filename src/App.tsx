import { Route, Routes } from 'react-router-dom';
import HelloComponent from './components/FirstScreen/FirstScreen';
import MainScreen from './components/MainScreen/MainScreen';
import UserCard from './components/User/userCard';
import NoMatch from './components/404/NoMatch';

function App() {
    //{
//    "env": {
//        "browser": true,
//        "es2021": true
//    },
//    "extends": [
////        "plugin:react/recommended",
////        "standard-with-typescript",
////        "airbnb-base",
////        "airbnb-typescript/base"
//    ],
//    "overrides": [
//    ],
//    "parserOptions": {
//        "ecmaVersion": "latest",
//        "sourceType": "module",
//        "project": "./tsconfig.json"
//    },
//    "plugins": [
//        "react"
//    ],
//    "rules": {
//        "import/prefer-default-export": "off",
//        "no-param-reassign": "off",
//        "import/extensions":"off",
//        "no-useless-catch": "off",
//        "@typescript-eslint/explicit-function-return-type": "off",
//        "@typescript-eslint/strict-boolean-expressions": "off",
//        "@typescript-eslint/no-misused-promises": "off",
//        "@typescript-eslint/no-floating-promises": "off",
//        "react/react-in-jsx-scope": "off",
//        "@typescript-eslint/no-non-null-assertion": "off",
//        "linebreak-style": 0
//    }
//}

    return (
        <Routes>
            <Route path='/' element={<HelloComponent/>}></Route>
            <Route path='/main' element={<MainScreen/>}></Route>
            <Route path='/user/:id' element={<UserCard/>}></Route>
            <Route path='*' element={<NoMatch/>}></Route>
        </Routes>
  );
}

export default App;
