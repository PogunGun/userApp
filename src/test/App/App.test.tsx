import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import App from '../../App';

describe('App Test', () => {
  test('should Main render', () => {
    const { asFragment } = render(
            <MemoryRouter initialEntries={['/error404']}>
                <App/>
            </MemoryRouter>,
    );
    expect(screen.getByTestId('not-found-page'))
      .toBeInTheDocument();
    expect(asFragment())
      .toMatchSnapshot();
  });
});
