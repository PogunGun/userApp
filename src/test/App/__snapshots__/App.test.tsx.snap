// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test should Main render 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-pw09vk"
    data-testid="not-found-page"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
      style="color: white;"
    >
      404
    </h1>
    <h6
      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
      style="color: white;"
    >
      The page you’re looking for doesn’t exist.
    </h6>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1ntrnb6-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Home
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
